### PR TITLE
Add Benoit's re commit dbt devblog

### PR DESCRIPTION
**Pull request by johnblust**
## Checklist
If you added new pages (delete if not applicable):
- [x] The page has been added to `website/sidebars.js`
- [x] The new page has a unique filename

### CODE REVIEW

**Comment by b-per** [comment]
Just a quick comment that we should remove "_(asciinema link we could put in the post https://asciinema.org/a/lTmefht77ZEr6kmP7DymaxjRF)_" or replace the image with the asciinema GIF

**Comment by johnblust** [comment]
Good catch, Benoit! I removed that comment.

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Optional wording suggestion: flip-flop these words:
```suggestion
pre-commit-dbt provides a range of tests that can be run on the models of our dbt project. In the case of a project which has existed for some time and which might not be entirely following best practices, I would recommend selecting a small subset of tests, fixing the project and adding more tests later on.
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Maybe hyperlink like this to make it easier to "jump to step 4"?
```suggestion
If we are using dbt Cloud, let’s jump to [step 4](#4-make-it-part-of-the-periodic-checks), where we will set up pre-commit-dbt as part of the CI process, otherwise, with dbt Core we can go to step 2.
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md:84]: 🤩

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Would you consider the following verbosity à la https://pip.pypa.io/en/stable/cli/pip_install/ ?

Same suggestion for any location with a bare `pip install [...]` 
```suggestion
Once in the Python virtual environment, installing `pre-commit` is as straightforward as running `python -m pip install pre-commit`.
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: I'm not sure what our company suggested words, phrases, and acronyms are. I think I saw Drew explicitly suggest using the phrase "data warehouse" over alternatives for Callum's metrics post.
```suggestion
For that reason, we will need our CI step to execute this command, which will require setting up a `profiles.yml` file providing dbt the information to connect to the data warehouse. Profiles files will be different for each data warehouse ([example here](https://docs.getdbt.com/reference/warehouse-profiles/snowflake-profile)).
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: I had to re-read this sentence a few times to understand it.

Maybe the following would be easier to understand?
```suggestion
We could setup rules that preventing any change to be merged if the GitHub action fails. Alternatively, this action step can be defined as merely informational. 
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Maybe an explicit link to the part of the documentation that explains `SKIP` and `--no-verify` flag?

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md:262]: You are a master at making compelling images, and WAY better at this than me.

Maybe something like this to highlight which steps are already covered and where they should focus next?
image

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Can you hyperlink to the Slack channels from the web, or no? If so, I love clickable links 😄

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Don't you love how programs automatically turn our `--` into `–`?!

```suggestion
In the case of a large project, the number of issues might be much bigger. If we use zsh as our shell, wildcard expansion can be used and we could run `pre-commit run --files models/mart/*` if we wanted to run all the checks only in the models stored under mart.
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: I hadn't heard of tangata before -- cool idea and it was fun to see the demo during the Coalesce video!

Unfortunately for me, it didn't install correctly for me out-of-the-box.

I did manage to get it installed by installing all of the following:
```
tangata
# needed to install and use tangata (see https://itsmycode.com/importerror-cannot-import-name-json-from-itsdangerous/)
itsdangerous==2.0.1
# needed to install and use tangata (see https://github.com/mjmeijer/donald_2021/issues/4)
werkzeug==2.0.3
# needed to install and use tangata
ruamel.yaml
```

But once I did get it installed and launched, it wasn't able to update my local files correctly. I didn't troubleshoot further from there.

So I'd cut out this entire section until the maintainer releases a new version 😢 

```suggestion
A great tool to update tests and descriptions for your models through a Web UI rather than updating dozens of YAML files is [tangata](https://github.com/ciejer/tangata_local), created by Chris Jenkins. You can watch Chris’s presentation at Coalesce [here](https://www.youtube.com/watch?v=9gFUmKyqe3w). 

Adding tests and descriptions can be done through a web browser and YAML files are automatically generated and updated in the background. We just need to commit and raise a PR!

![image of a dbt model documentation and description UI](/img/blog/2022-07-26-pre-commit-dbt/customers-model.png)
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: It would be awesome if you could suggest a default file name for those of us that are unimaginative and/or lazy (me! 🙋).
```suggestion
Finally, we can create a new YAML file to define our GitHub action. e.g. `.github/workflows/pre_commit_checks.yml` The name is not important but this file must be saved  under the folders `.github/workflows/` (create those if they don’t exist yet)
```

**Review by dbeatty10 — CHANGES_REQUESTED** [review]
@b-per this is really top-notch and very educational. I had a blast learning and trying it out.

I used [this](https://github.com/dbt-labs/jaffle_shop_duckdb/pull/30) pull request to try things out -- nearly everything worked flawlessly. Comments note in the few places that didn't work for me.

My main recommendations:
- Remove tāngata since I couldn't get it to work 😢
- Consider if more text can be formatted as `code` or `code_related_names` -- there seemed to be quite a few places that I personally normally format with backticks

I'm going to "request changes", but feel free to take or leave any of my suggestions. I feel good about approving once you give those an initial look.

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md]: I will remove the part about tangata from the code. Thanks for checking it!

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md]: I will give it a go!

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: I will check it tomorrow. Thanks for the suggestion!

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md]: I am personally not doing it 😁 but I am happy to update the instructions to replace `pip` by `python -m pip`

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md]: Much better!

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md]: Will do! Thanks for the feedback

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [reply website/blog/2022-07-26-pre-commit-dbt.md]: In case you haven't seen any explanations _why_ `python -m pip` is the recommended command:
- Longer:[ Why you should use `python -m pip`](https://snarky.ca/why-you-should-use-python-m-pip/)
- Shorter:[ Use 'python -m pip' Everywhere](https://adamj.eu/tech/2020/02/25/use-python-m-pip-everywhere/)

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Another slight update I missed before:
```suggestion
We could set up rules that prevent any change to be merged if the GitHub action fails. Alternatively, this action step can be defined as merely informational. 
```

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (dbeatty10) [inline website/blog/2022-07-26-pre-commit-dbt.md]: Why did the [asciinema](https://asciinema.org/) get dropped? It's nice to see the commands in action. Even cooler once people know that they can copy-n-paste from it!

https://asciinema.org/a/lTmefht77ZEr6kmP7DymaxjRF

File: website/blog/2022-07-26-pre-commit-dbt.md
Comment (b-per) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: Thanks @dbeatty10 
I followed what you suggested and just used the same style for the diagram. I don't know how to update the image though. Could you help?

image

Comment (johnblust) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: I can get that image updated for you! I'll comment here when I've committed the change

Comment (johnblust) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: Okay, the new image should be reflected in the content now!

Comment (dbeatty10) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: Looks awesome @b-per ! Thanks @johnblust !

Comment (johnblust) [reply website/blog/2022-07-26-pre-commit-dbt.md:262]: Cool, if we feel like this is good to go, can we get approval @dbeatty10?

**Review by dbeatty10 — APPROVED** [review]
Stellar work @b-per + @johnblust 🥇

**Comment by johnblust** [comment]
@b-per If this looks good to you, let me know and I'll update the publication date and get this merged! Social promotion is still set for this Friday!

**Comment by b-per** [comment]
All good from my side as well!

**Comment by dbeatty10** [comment]
@b-per:
> Just a quick comment that we should remove "_(asciinema link we could put in the post https://asciinema.org/a/lTmefht77ZEr6kmP7DymaxjRF)_" or replace the image with the asciinema GIF

@johnblust 
> Good catch, Benoit! I removed that comment.

I meant to ask this earlier:

Is there a reason why we aren't embedding the asciinema GIF in the post? asciinema does two cool things:
1. Shows what the commands and output looks like in action
1. Allows the commands to be copied-n-pasted

Obviously not a show-stopper to leave it out, just something that is kinda cool if we can get it in.

**Comment by b-per** [comment]
No reason except that embedding requires a `<script></script>` tag (see [here](https://asciinema.org/docs/embedding)) and I don't think that we can add it in Markdown. If it is technically possible to somehow embed it rather than have it as an image + external link this would definitely be better!

**Comment by dbeatty10** [comment]
I see! Thank you for explaining @b-per 🙌 

The image + external link is a good solution given that context.

### Confirmed with experimentation
I experimented with embedding via the `<script>` tag per asciinema's documentation and confirmed it didn't work.

Quick google search didn't yield any quick-n-easy solutions. Just areas to dig into further like:
- https://github.com/facebook/docusaurus/issues/3980
- https://stackoverflow.com/questions/63810181/docusaurus-2-run-custom-script-for-every-page

**Pull request merged**
